packages/astro/src/core/render/route-cache.ts

```diff
@@ -28,39 +29,39 @@ export async function callGetStaticPaths({
 	route,
 	ssr,
 }: CallGetStaticPathsOptions): Promise<RouteCacheEntry> {
-	validateDynamicRouteModule(mod, { ssr, logging });
+	validateDynamicRouteModule(mod, { ssr, logging, route });
 	// No static paths in SSR mode. Return an empty RouteCacheEntry.
 	if (ssr) {
 		return { staticPaths: Object.assign([], { keyed: new Map() }) };
 	}
-	// Add a check here to my TypeScript happy.
+	// Add a check here to make TypeScript happy.
 	// This is already checked in validateDynamicRouteModule().
 	if (!mod.getStaticPaths) {
 		throw new Error('Unexpected Error.');
 	}
+
 	// Calculate your static paths.
 	let staticPaths: GetStaticPathsResult = [];
-	staticPaths = (
-		await mod.getStaticPaths({
-			paginate: generatePaginateFunction(route),
-			rss() {
-				throw new Error(
-					'The RSS helper has been removed from getStaticPaths! Try the new @astrojs/rss package instead. See https://docs.astro.build/en/guides/rss/'
-				);
-			},
-		})
-	).flat();
+	staticPaths = await mod.getStaticPaths({
+		paginate: generatePaginateFunction(route),
+		rss() {
+			throw new AstroError(AstroErrorData.GetStaticPathsDeprecatedRSS);
+		},
+	});
+
+	if (isValidate) {
+		validateGetStaticPathsResult(staticPaths, logging, route);
+	}
 
+	staticPaths = staticPaths.flat();
```

Comment:
Previously, `validateGetStaticPathsResult` (which validates the shape of the return value of `getStaticPaths`) would run after `stringifyParams` (which validates the shape of the individual entries of the return value of `getStaticPaths`)

So, if the return value was wrong, the build would just fail with a cryptic error in most cases as it would try to read entries that don't exist. Inverting the calls makes more sense